credential/signature.go

```diff
@@ -0,0 +1,110 @@
+package credential
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/pkg/errors"
+)
+
+// VerifyCredentialSignature verifies the signature of a credential of any type
+// TODO(gabe) support other types of credentials https://github.com/TBD54566975/ssi-sdk/issues/352
+func VerifyCredentialSignature(ctx context.Context, genericCred any, resolver did.Resolver) (bool, error) {
+	if genericCred == nil {
+		return false, errors.New("credential cannot be empty")
+	}
+	if resolver == nil {
```

Comment:
if the resolver was empty could there be a default one we load?

Reply:
I'm not sure if we should, because that would imply supporting DID methods that the caller may not want to resolve